### ISSUE TITLE
smooth=true for log axes

### ISSUE THREAD

**Issue by aahaselgrove** [JuliaPlots/Plots.jl#2227]
When using log axes (i.e. axis=:log), I would expect smooth=true to give a visually smooth line through the points (i.e. a power regression). Currently, it gives a linear regression regardless of axis settings, just displaying another line on the plot.

````
using Plots(); pyplot()

x = [3, 4, 5, 6, 7]
y = [9, 16, 25, 36, 49]
plot(x, y, marker=:circle, smooth=true)
````
![Figure_1](https://user-images.githubusercontent.com/2556805/67258948-7e42da80-f4d6-11e9-951e-9ca807322a50.png)

````
using Plots(); pyplot()

x = [3, 4, 5, 6, 7]
y = [9, 16, 25, 36, 49]
plot(x, y, marker=:circle, axis=:log, smooth=true)
````
![Figure_2](https://user-images.githubusercontent.com/2556805/67258949-7e42da80-f4d6-11e9-8ca3-6e6151294d81.png)

**Comment by mkborregaard** [comment]
I agree - an implementation should then force a recalculation of the smoother series whenever the scales are changed.
